stdlib.sh

```diff
@@ -251,6 +269,52 @@ export std_ERRNO=0
 declare -a __STDLIB_OWNED_FILES
 
 declare std_INTERNAL_DEBUG="${SLDEBUG:-0}"
+
+# Command controlled execution
+export STDLIB_EXEC_COMMAND=""
+export STDLIB_EXEC_RESULT=""
+export STDLIB_EXEC_STATUS=0
+
+readonly STDLIB_EXEC_WARNONERR="w"
+readonly STDLIB_EXEC_ERRONERR="k"
+readonly STDLIB_EXEC_FAILONERR="f"
+
+
+## debug verbosity
+readonly STDLIB_DEBUG_NORMAL=1
+readonly STDLIB_DEBUG_SSH=2
+readonly STDLIB_DEBUG_RSYNC=3
+readonly STDLIB_DEBUG_REMOTE=4
+readonly STDLIB_DEBUG_JSON=5
+readonly STDLIB_DEBUG_PARALLEL=6
+readonly STDLIB_DEBUG_FUNCTION=7
+
+## command stream capture
+readonly STDLIB_STREAM_NORMAL=0
+readonly STDLIB_STREAM_STDOUT=1
+readonly STDLIB_STREAM_STDERR=2
+readonly STDLIB_STREAM_ALL=3
+readonly STDLIB_STREAM_NONE=4
+readonly STDLIB_STREAM_INTERACTIVE=5
+
+## Colored output
+declare std_COLOUR_MAP_FILE="${STDLIB_COLOUR_MAP:-/usr/local/lib/stdlib.colours}"
+declare std_COLOUR_MAP_SECTION="colours"
+
+## Colourisation is off by default
+declare std_COLOUR_START_INFO=""
+declare std_COLOUR_START_NOTICE=""
+declare std_COLOUR_START_DEBUG=""
+declare std_COLOUR_START_EXEC=""
+declare std_COLOUR_START_OK=""
+declare std_COLOUR_START_WARNING=""
+declare std_COLOUR_START_FATAL=""
+declare std_COLOUR_START_FAIL=""
+declare std_COLOUR_START_ERROR=""
+declare std_COLOUR_END=""
+
+declare std_COLOUR_ON="${STDLIB_WANT_COLOURISATION:-0}"
```

Comment:
Rather than adding this significant amount of constant global state, could this data be collapsed into a string or function to be `eval`d when needed?

Rather than creating an additional `COLOUR_ON` variable, could `STDLIB_WANT_COLOURISATION` be used in its place?